tests/functional/deploy-cluster.yml

```diff
@@ -0,0 +1,280 @@
+---
+- name: gather facts and prepare system for cephadm
+  hosts:
+    - "{{ mon_group_name }}"
+    - "{{ osd_group_name }}"
+    - "{{ mds_group_name }}"
+    - "{{ rgw_group_name }}"
+    - "{{ mgr_group_name }}"
+    - "{{ rbdmirror_group_name }}"
+    - "{{ nfs_group_name }}"
+    - "{{ iscsi_gw_group_name }}"
+    - "{{ monitoring_group_name }}"
+  become: true
+  gather_facts: false
+  tasks:
+    - import_role:
+        name: ceph-defaults
+
+    - name: gather and delegate facts
+      setup:
+        gather_subset:
+          - 'all'
+          - '!facter'
+          - '!ohai'
+      delegate_to: "{{ item }}"
+      delegate_facts: true
+      with_items: "{{ groups['all'] }}"
+      run_once: true
+      when: delegate_facts_host | bool
+
+    - name: check if podman binary is present
+      stat:
+        path: /usr/bin/podman
+      register: podman_binary
+
+    - name: set_fact container_binary
+      set_fact:
+        container_binary: "{{ 'podman' if (podman_binary.stat.exists and ansible_facts['distribution'] == 'Fedora') or (ansible_facts['os_family'] == 'RedHat' and ansible_facts['distribution_major_version'] == '8') else 'docker' }}"
```

Comment:
```suggestion
    - name: set_fact container_binary
      set_fact:
        container_binary: "{{ 'podman' if podman_binary.stat.exists else 'docker' }}"
```
?

Reply:
in fact, are we going to use docker instead of podman in the ci? (this playbook is here for CI testing only)